src/hdmf/backends/hdf5/h5_utils.py

```diff
@@ -177,8 +178,9 @@ def read_spec(self, spec_path):
         return self.__read(spec_path)
 
     def read_namespace(self, ns_path):
-        ret = self.__read(ns_path)
-        ret = ret['namespaces']
+        if self.__cache is None:
+            self.__cache = self.__read(ns_path)
+        ret = self.__cache['namespaces']
```

Comment:
Note to self: caching of the spec by the H5SpecReader should be mentioned in the release notes